src/canvasMachine.tsx

```diff
@@ -178,6 +182,25 @@ export const canvasMachine = createMachine<typeof canvasModel>({
         return position;
       }),
     },
+    'elkGraph.UPDATE': {
+      actions: canvasModel.assign({
+        elkGraph: (_, e) => e.elkGraph,
+      }),
+    },
+    ZOOM_TO_FIT: {
+      actions: canvasModel.assign({
+        zoom: (ctx) => {
+          return Math.min(
+            ctx.canvasPanelPosition.width / ctx.elkGraph!.width!,
+            ctx.canvasPanelPosition.height / ctx.elkGraph!.height!,
+          );
```

Comment:
With a very simple machine of:
```js
import { createMachine } from "xstate";

createMachine({
  initial: "a",
  states: {
    a: {},
  },
});
```
And values like:
```js
// full window
{
  canvasPanelPosition: { offsetX: 0, offsetY: 0, width: 1120, height: 826 },
  elkGraph: { width: 85.125, height: 140 }
}

// with debugger opened
{
  canvasPanelPosition: { offsetX: 0, offsetY: 0, width: 1120, height: 477 },
  elkGraph: { width: 85.125, height: 140 }  
}
```
We get smth like:
<img width="1678" alt="Screenshot 2021-08-25 at 16 21 09" src="https://user-images.githubusercontent.com/9800850/130808949-6ebe7fd2-6e1b-4c7f-919a-7434200ecd6f.png">

With another small machine (just entry action, without states), the result can look even more off:
<img width="1678" alt="Screenshot 2021-08-25 at 16 20 14" src="https://user-images.githubusercontent.com/9800850/130809040-e698b24d-08ea-44a5-a309-cd030a5adc1c.png">